Content.Shared/Physics/MoverController.cs

```diff
@@ -12,17 +13,29 @@ public class MoverController : VirtualController
 
         public void Move(Vector2 velocityDirection, float speed)
         {
-            if (ControlledComponent?.Owner.IsWeightless() ?? false)
-            {
+            if (ControlledComponent == null)
+                return;
+
+            if (ControlledComponent.Owner.IsWeightless())
+                return;
+
+            if (!ControlledComponent.OnGround)
                 return;
-            }
 
             Push(velocityDirection, speed);
         }
 
         public void Push(Vector2 velocityDirection, float speed)
         {
-            ControlledComponent.Force += velocityDirection * speed * 5000;
+            if (ControlledComponent == null) return;
+            if (ControlledComponent.LinearVelocity.Length > speed) return;
+            ControlledComponent.Force += velocityDirection * speed * ControlledComponent.Mass;
+        }
+
+        public void StopMoving()
+        {
+            if (ControlledComponent == null || !ControlledComponent.OnGround) return;
+            ControlledComponent.Force += -ControlledComponent.LinearVelocity * ControlledComponent.Mass;
```

Comment:
this will make the controlled component stop over one second no matter the speed, 
also ignores any force that might be acting on the component, there should be a specific stopping force the mover can exhibit against current velocity. i.e. -LinerVelocity.Normalized * walkSpeed * Mass / frameTime to be able to stop walking in single frame but not sprinting/ larger external forces.